### PR TITLE
Jg/text module

### PR DESCRIPTION
**Pull request by jaimeglez**

### CODE REVIEW

**Comment by jaimeglez** [comment]
You can see the page of module catalog in [/module-catalog](https://e39e977a.sideways.pages.dev/module-catalog)

File: content/pages/module-catalog.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+tags:
+  - document
+modules:
+  - _type: text-module
+    text1: Culture Defining
+    text2: Creative
+    text3: for
+    text4: Modern Hospitality
+    divider: true
+    secondaryText: Sideways is a digital-first branding and creative agency uniquely positioned to differentiate your brand in a world where branding never stops.
```
Comment (luishdz1010) [inline content/pages/module-catalog.mdx:15]: How does the field work? I don't see any definition of text1-text4 anywhere. In other words, where are the ACFs?

File: package.json [hunk 2]
```diff
@@ -35,6 +35,7 @@
     "react-schemaorg": "^2.0.0",
     "react-twitter-embed": "^4.0.4",
     "react-youtube": "^10.0.0",
+    "sass": "^1.55.0",
```
Comment (luishdz1010) [inline package.json:38]: Not needed I believe (already on dev-deps)

File: src/components/Text.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    <div className={`text-module ${className ? className : ''}`}>
+      <div className="container">{primaryContent}</div>
+      {divider && <hr />}
+      <div className="container">
+        {secondaryContent}
+
+        {cta}
+      </div>
+
+      <style jsx>{`
```
Comment (luishdz1010) [inline src/components/Text.tsx:22]: Use scss file instead (also don't hardcode media queries)

File: src/components/Text.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  primaryContent: ReactNode
+  divider?: boolean
+  secondaryContent?: ReactNode
+  cta?: ReactNode
+  className?: string
+}
+
+export function Text({ primaryContent, divider, secondaryContent, cta, className }: Props) {
+  return (
+    <div className={`text-module ${className ? className : ''}`}>
```
Comment (luishdz1010) [inline src/components/Text.tsx:13]: `className ?? ''`

File: src/components/Text.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+type Props = {
+  primaryContent: ReactNode
+  divider?: boolean
+  secondaryContent?: ReactNode
+  cta?: ReactNode
+  className?: string
+}
+
+export function Text({ primaryContent, divider, secondaryContent, cta, className }: Props) {
```
Comment (luishdz1010) [inline src/components/Text.tsx:11]: Text is too generic of a name, find a better, larger one

File: src/modules/TextModule.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  text4: string
+  divider: boolean
+  secondaryText: string
+  cta: {
+    href: string
+    text: string
+  }
+}
+
+export default function TextModule({
```
Comment (luishdz1010) [inline src/modules/TextModule.tsx:19]: This name is better, but if you can find a better name that's also good (assuming we have other similar single-text modules elsewhere)

File: src/modules/text.module.scss [hunk 7]
```diff
@@ -0,0 +1,145 @@
+@import '../style/lib';
+@import '../style/modules';
+
```
Comment (luishdz1010) [inline src/modules/text.module.scss:3]: Remember to use the module mixin at the root of the component. Also, all modules need a root element of type `section` so we can use  ` :first/last-of-type` selectors for overriding spacing.

File: src/style/lib/typography.scss [hunk 8]
```diff
@@ -11,10 +11,65 @@
   text-transform: uppercase;
 }
 
+@mixin t-Oscar_Light_XXLarge_White() {
+  color: $white;
+  font-weight: 100;
```
Comment (luishdz1010) [inline src/style/lib/typography.scss:16]: Is this font weight actually in the font files?

**Review by luishdz1010 — CHANGES_REQUESTED** [review]

File: src/style/lib/typography.scss (hunk 8 above)
Comment (jaimeglez) [reply src/style/lib/typography.scss:16]: my bad, that was to be intended 300, changing

File: src/components/TextDividerText.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import React, { ReactNode } from 'react'
+import styles from '../style/text-divider-text.module.scss'
+
+type Props = {
+  primaryContent: ReactNode
+  divider?: boolean
+  secondaryContent?: ReactNode
+  cta?: ReactNode
```
Comment (luishdz1010) [inline src/components/TextDividerText.tsx:8]: Can be merged with secondaryContent

File: src/modules/HeadlineIntro.module.scss [hunk 10]
```diff
@@ -1,6 +1,25 @@
 @import '../style/lib';
 @import '../style/modules';
 
+.root {
+  @include module();
+  @include module-spacing-1();
+
+  // same as adding class "container" or using <Container>
+  // @include make-container();
```
Comment (luishdz1010) [inline src/modules/HeadlineIntro.module.scss:9]: remove

File: src/pages/[page].tsx [hunk 11]
```diff
@@ -62,6 +62,8 @@ export const getStaticProps: GetStaticProps = async ({ params }) => {
     engines: { yaml: (s) => yaml.load(s, { schema: yaml.JSON_SCHEMA }) as object },
   })
 
+  console.log('data-->', data)
```
Comment (luishdz1010) [inline src/pages/[page].tsx:65]: x

File: src/components/TextDividerText.tsx [hunk 12]
```diff
@@ -0,0 +1,30 @@
+import React, { ReactNode } from 'react'
+import styles from '../style/text-divider-text.module.scss'
```
Comment (luishdz1010) [inline src/components/TextDividerText.tsx:2]: This file should be co-located in the same folder as the component

File: src/style/text-divider-text.module.scss [hunk 13]
```diff
@@ -0,0 +1,31 @@
+@import '../style/lib';
+@import '../style/modules';
```
Comment (luishdz1010) [inline src/style/text-divider-text.module.scss:2]: Not needed I think

File: src/style/text-divider-text.module.scss [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+  height: 60px;
+  border: none;
+  background: $white;
+  margin: 22px auto;
+}
+
+@media (min-width: 576px) {
+  .text-module {
+    padding: 110px 0;
+  }
```
Comment (luishdz1010) [inline src/style/text-divider-text.module.scss:24]: Is this used? Seems weird to call it `module` when this is just a component

File: content/pages/module-catalog.mdx [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+date: 2022-05-12
+author: jaime-glez
+tags:
+  - document
+modules:
+  - _type: headline-intro-module
+    text1: Culture Defining
+    text2: Creative
+    text3: for
+    text4: Modern Hospitality
```
Comment (luishdz1010) [inline content/pages/module-catalog.mdx:13]: Remove

File: package.json [hunk 16]
```diff
@@ -48,6 +51,7 @@
     "eslint-config-next": "^12.3.1",
     "eslint-config-prettier": "^8.5.0",
     "jest": "^29.1.2",
+    "netlify-cms-proxy-server": "^1.3.24",
```
Comment (luishdz1010) [inline package.json:54]: This is added twice

File: src/modules/HeadlineIntroModule.tsx [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+        divider={divider}
+        primaryContent={
+          <h2 className={styles['headline']}>
+            <div className={styles['headline-2']}>
+              {text1}
+              <span className={styles['line']} />
+            </div>
+            <div className={styles['headline-3']}>
+              {text2}
+              <span className={styles['line']} />
```
Comment (luishdz1010) [inline src/modules/HeadlineIntroModule.tsx:40]: Here you can hardcode the strings that were on the yml. Or add then as default params of the function `text1 = 'lorem', text2 ='etc'`

File: src/pages/[page].tsx [hunk 18]
```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import { GetStaticPaths, GetStaticProps } from 'next'
+import matter from 'gray-matter'
+import { fetchPageContent, PageContent } from '../lib/pages'
+import fs from 'fs'
+import yaml from 'js-yaml'
+import { parseISO } from 'date-fns'
+import PageLayout from '../components/PageLayout'
+import { MDXRemoteSerializeResult } from 'next-mdx-remote/dist/types'
```
Comment (luishdz1010) [inline src/pages/[page].tsx:9]: probably not a good idea to import from dist/types. check if the import is available the root of the pkg

**Review by luishdz1010 — APPROVED** [review]

**Pull request merged**